### PR TITLE
Add functionality to exclude soft deleted records by default

### PR DESCRIPTION
**Pull request by Dacello**
connects to https://github.com/revelrylabs/ecto_soft_delete/issues/78

Description:
- Adds a `prepare_query` method to the soft delete repo module which overrides all query operations to exclude soft deleted records if the given schema contains `deleted_at`

### CODE REVIEW

**Review by bryanjos — APPROVED** [review]
Nice!

**Comment by Dacello** [comment]
@bryanjos thanks -- got major help @NOrzechowski and @Samkeer1 on this one. Still waiting to get tests to pass locally

**Comment by Dacello** [comment]
@bryanjos tests fixed, but can you take another look? Had to make some updates to use `utc_datetime_usec` to make it compatible with Ecto 3

**Comment by bryanjos** [comment]
@Dacello still looks good. The change in the datatype will cause issues for previous users of it though

**Comment by Dacello** [comment]
@bryanjos yea for sure, we're gonna want to bump it up a major version i think

**Review by NOrzechowski — CHANGES_REQUESTED** [review]
Should add a changelog entry

**Review by NOrzechowski — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,11 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
+## [2.0.0] - 2020-05-06
+
+- Exclude soft deleted records by default in `Ecto.SoftDelete.Repo`
+- Make `soft_delete_fields` use `utc_datetime_usec` instead of `utc_datetime`
```
Comment (oohnoitz) [inline CHANGELOG.md]: nit: let's indicate that this is a breaking change if its gonna cause issues.

```
- **BREAKING**: make ...
```

**Review by oohnoitz — APPROVED** [review]

**Comment by Dacello** [comment]
:shipit:

**Pull request merged**